src/commands/thread.ts

```diff
@@ -132,41 +133,65 @@ export default command({
-                            new ButtonBuilder()
-                                .setCustomId('reopen')
-                                .setLabel('Mark as Unsolved')
-                                .setStyle(ButtonStyle.Secondary)
-                                .setEmoji('❔'),
+                    if (HELP_THREAD_CHANNELS.includes(thread.parentId)) {
+                        // Attempt to solve the thread
+                        await solve_thread(
+                            thread,
+                            interaction.member as GuildMember,
                         );
-                    msg.components = [row];
-                    await bot_message.edit(msg);
+                        // Successfully solved the thread
+                        // Get the first message in the thread
+                        const start_message = await thread.fetchStarterMessage();
+                        // Get the first 2 messages after the start message
+                        const messages = await thread.messages.fetch({
+                            limit: 2,
+                            after: start_message.id,
+                        });
+                        // Filter the messages to find the bot message with the buttons
+                        const bot_message = messages
+                            .filter((m) => m.components.length > 0)
+                            .first() as Message;
+                        // Change the message
+                        const msg = wrap_in_embed(
+                            'Thread solved. Thank you everyone! 🥳',
+                        ) as MessageEditOptions;
+                        // Change the button
+                        const row =
+                            new ActionRowBuilder<ButtonBuilder>().addComponents(
+                                new ButtonBuilder()
+                                    .setCustomId('reopen')
+                                    .setLabel('Mark as Unsolved')
+                                    .setStyle(ButtonStyle.Secondary)
+                                    .setEmoji('❔'),
+                            );
+                        msg.components = [row];
+                        await bot_message.edit(msg);
+                        // Commands require a reply
+                        await interaction.followUp(wrap_in_embed('Thread solved.'));
+                        // Delete the reply after 10 seconds
+                        setTimeout(async () => {
+                            await interaction.deleteReply();
+                        }, 10000);
+                    }
+                    // Not a help channel, check if its parent is a ForumChannel
+                    if (!(thread.parent instanceof ForumChannel))
+                        throw new Error("Can't solve a non-help channel");
+                    // Parent forum channel
+                    const solveChannel = thread.guild.channels.cache.get(thread.parentId) as ForumChannel
```

Comment:
I don't think we're filtering for which forums should be managed?

Reply:
We don't limit it to specific channels no. We can, but currently it's set to only fetch 100 messages from ThreadChannel's that are not archived. That shouldn't be THAT many threads to go through. Then it fetches messages from each text channel. Imo there's no reason to complicate things with limiting which exact channels to fetch, I haven't seen any insane resource usage as a result of the current setup, at most I'd say I guess we can remove the text channels entirely if we want since currently there shouldn't be any messages that get fixed by it. It's more of a future proofing so I don't get a headache later when I fix another issue.